### PR TITLE
Delta client get secret #55

### LINKED ISSUE #53: Client Share Secret functionality
Create a derived secret using another identity's public key.

Depends on ability to get secret.

### PR DESCRIPTION
**Pull request by numbat**
Fixes #53, #55

### CODE REVIEW

File: src/main/python/covata/delta/apiclient.py [hunk 1]
```diff
@@ -270,7 +267,7 @@ def get_secret_content(self, requestor_id, secret_id):
         :param str requestor_id: the authenticating identity id
         :param str secret_id: the secret id to be retrieved
         :return: the retrieved secret
-        :rtype: bytes
+        :rtype: dict[str, str]
         """
```
Comment (myuwono) [inline src/main/python/covata/delta/apiclient.py]: Shouldn't this be a string? Maybe `response.text` is more appropriate for this method

Comment (numbat) [inline src/main/python/covata/delta/apiclient.py]: It is a string isn't it?

Comment (numbat) [inline src/main/python/covata/delta/apiclient.py]: Nevermind - yes cut and paste error on the method rtype.

**Comment by myuwono** [comment]
should we update docs to include the Secret fluent object?

**Comment by numbat** [comment]
Yes we should - I keep forgetting the rst.

I'll add to this PR before merge.

File: src/main/python/covata/delta/apiclient.py [hunk 2]
```diff
@@ -83,7 +81,7 @@ def get_identity(self, requestor_id, identity_id):
         :param str requestor_id: the authenticating identity id
         :param str identity_id: the identity id to retrieve
         :return: the retrieved identity
-        :rtype: dict[str, any]
+        :rtype: dict[str, str]
```
Comment (myuwono) [inline src/main/python/covata/delta/apiclient.py]: are you sure with the return type? if i remember correctly, get identity returns a metadata dict in the json which throws off that `dict[str, str]` type. Another approach is we can return them separately as a tuple of the identity dict and identity metadata dict `(dict[str, str], dict[str, str])`, that will also work but not sure of its correctness..

File: src/main/python/covata/delta/apiclient.py [hunk 3]
```diff
@@ -229,7 +225,7 @@ def get_secret(self, requestor_id, secret_id):
         :param str requestor_id: the authenticating identity id
         :param str secret_id: the secret id to be retrieved
         :return: the retrieved secret
-        :rtype: dict[str, any]
+        :rtype: dict[str, str]
```
Comment (myuwono) [inline src/main/python/covata/delta/apiclient.py]: here, can't remember exactly but I'm pretty sure there's also a nested `dict` of encryption details in the response json. same case with identity response json.

Comment (numbat) [inline src/main/python/covata/delta/apiclient.py]: Oh - good pickup, yes I'll review those ones, pretty sure you are correct.

It doesn't matter if they remain dict[str,any] anyway even if it was dict[str, str]

Comment (myuwono) [inline src/main/python/covata/delta/apiclient.py]: Nice, yeah the only type we can be sure of is the key's which is a str. All good man, thanks!

**Comment by myuwono** [comment]
Looks good! 👍

**Pull request merged**